borg/remote.py

```diff
@@ -164,11 +167,21 @@ def __init__(self, location, create=False, lock_wait=None, lock=True, args=None)
         self.id = self.call('open', location.path, create, lock_wait, lock)
 
     def __del__(self):
-        self.close()
+        if self.p:
+            self.close()
+            assert False, "cleanup happened in Repository.__del__"
```

Comment:
Ah, of course, in the tests tearDown performs "hard" tear downs and just rmtree()s the repository. That probably raced with `__del__` and lead to the locks being deleted before del had a change to clean them up.